lib/product.rb

```diff
@@ -0,0 +1,33 @@
+class Product
+  attr_reader :category,
+              :name,
+              :quantity,
+              :unit_price
+
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity
+  end
+
+  def quantity
+    @quantity.to_i
+  end
```

Comment:
Since you are making a method here for quantity, I'd get rid of the attr_reader for quantity. The attr_reader will read the quantity on line 11, but you want it so that when you call `.quantity` that you have it returned as an integer.